spec.html

```diff
@@ -734,6 +745,283 @@ <h1>Example Source Text Module Record Graphs</h1>
+    </p>
+
+    <emu-table id="table-module-graph-cycle-async-fields-2" caption="Module fields after module _E_ finishes executing">
+      <table>
+        <thead>
+          <tr>
+            <th>Module</th>
+            <th>[[DFSIndex]]</th>
+            <th>[[DFSAncestorIndex]]</th>
+            <th>[[AsyncEvaluating]]</th>
+            <th>[[AsyncParentModules]]</th>
+            <th>[[PendingAsyncDependencies]]</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <th>_C_</th>
+            <td>2</td>
+            <td>0</td>
+            <td>*true*</td>
+            <td>&laquo; _A_ &raquo;</td>
+            <td>1 (_D_)</td>
+          </tr>
+          <tr>
+            <th>_E_</th>
+            <td>4</td>
+            <td>4</td>
+            <td>*false*</td>
+            <td>&laquo; _C_ &raquo;</td>
+            <td>0</td>
+          </tr>
+        </tbody>
+      </table>
+    </emu-table>
+
+    <p>
+      _D_ is next to finish (as it was the only module that was still executing).
+      When that happens, AsyncModuleExecutionFulfilled is called again and _D_.[[AsyncEvaluating]] is set to *false*.
+      Both _B_.[[PendingAsyncDependencies]] and _C_.[[PendingAsyncDependencies]] are decremented to become 0, so ExecuteAsyncModule is called and they start executing.
+      The fields of the updated modules are as given in <emu-xref href="#table-module-graph-cycle-async-fields-3"></emu-xref>.
+    </p>
+
+    <emu-table id="table-module-graph-cycle-async-fields-3" caption="Module fields after module _D_ finishes executing">
+      <table>
+        <thead>
+          <tr>
+            <th>Module</th>
+            <th>[[DFSIndex]]</th>
+            <th>[[DFSAncestorIndex]]</th>
+            <th>[[AsyncEvaluating]]</th>
+            <th>[[AsyncParentModules]]</th>
+            <th>[[PendingAsyncDependencies]]</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <th>_B_</th>
+            <td>1</td>
+            <td>0</td>
+            <td>*true*</td>
+            <td>&laquo; _A_ &raquo;</td>
+            <td>0</td>
+          </tr>
+          <tr>
+            <th>_C_</th>
+            <td>2</td>
+            <td>0</td>
+            <td>*true*</td>
+            <td>&laquo; _A_ &raquo;</td>
+            <td>0</td>
+          </tr>
+          <tr>
+            <th>_D_</th>
+            <td>3</td>
+            <td>0</td>
+            <td>*false*</td>
+            <td>&laquo; _B_, _C_ &raquo;</td>
+            <td>0</td>
+          </tr>
+        </tbody>
+      </table>
+    </emu-table>
+
+    <p>
+      Let us assume that _C_ finishes executing next.
+      When that happens, AsyncModuleExecutionFulfilled is called again, _C_.[[AsyncEvaluating]] is set to *false* and _A_.[[PendingAsyncDependencies]] is decremented to become 1.
+      The fields of the updated modules are as given in <emu-xref href="#table-module-graph-cycle-async-fields-4"></emu-xref>.
+    </p>
```

Comment:
Instead of a valid completion, it might be quite nice to introduce an error into the algorithm example at this point, as error propogation is an important part of the algorithm. No pressure to do this, just a suggestion.

Reply:
Filed #118.